flink-core/src/main/java/org/apache/flink/api/common/ExecutionConfig.java

```diff
@@ -90,6 +90,8 @@
 	/** Defines how data exchange happens - batch or pipelined */
 	private ExecutionMode executionMode = ExecutionMode.PIPELINED;
 
+	private ScheduleMode scheduleMode = ScheduleMode.LAZY_FROM_SOURCES;
```

Comment:
Does this mean if users do not specify the `ScheduleMode `, even stream jobs will be executed in batch style?

Reply:
Good point!
It changes default behavior of `DataStream` job, `EAGER` -> `LAZY_FROM_SOURCE`.
Should we move `ScheduleMode` from `ExecutionConfig` to `StreamExecutionEnvironment` since `ExecutionConfig` is also used by `DataSet`? `DataSet` need `LAZY_FROM_SOURCE`, `DataStream` need `EAGER` by default.

Reply:
Ha! you got me! I left this in on purpose because I wanted to see if all the tests still pass with this different default. We should of course not have this in the final code.

I put it in `ExecutionConfig` because it is available in `flink-core` while `StreamExecutionEnvironment` is not. I had a feeling that maybe the Table API will need access to this at some point or other. What do you think?

Reply:
Changing it to EAGER by default may have `DataSet` job get scheduled in streaming style, which might be unexpected. Shall we override it to be LAZY_FROM_SOURCES by default for DataSet?

Reply:
I have a concern that `DataSet` would ignore this `ScheduleMode` setting. It looks a bit imperfect. Unless you think we should not care about `DataSet` anymore? 

Regard to the `flink-core` part, it's an interesting point that I didn't notice. Yes, I agree with you. I also have a strong feeling that Table API will need it sooner or later. I was thinking Table API may generate `StreamGraph` itself in the future. In that way, Table API could set this property directly. But it's hard to say what it will look like.

Reply:
So moving it to `StreamExecutionEnvironment` might make sense after all, and not have it in `ExecutionConfig`, I think. (Still thinking ...)